### PR TITLE
Move python-click to SPECS folder and bump version to 8.0.4 based on fedora 36 imprt.

### PR DESCRIPTION
**Pull request by Sumynwa**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Moving python-click to SPECS folder.
Needed as run dependency by cassandra medusa.

###### Change Log  
- Move python-click Spec from SPECS-EXTENDED to SPECS folder.
- Bump version to 8.0.4 based on Fedora 36.
- Needed as run dependency for cassandra medusa.

###### Does this affect the toolchain?  
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/python-click/python-click.spec [hunk 1]
```diff
@@ -30,132 +20,214 @@ comes with good defaults out of the box.
...
-%files -n python%{python3_pkgversion}-%{pypi_name}
+%files -n python%{python3_pkgversion}-%{pypi_name} -f %pyproject_files
 %license LICENSE.rst
 %doc README.rst CHANGES.rst
-%{python3_sitelib}/%{pypi_name}/
-%{python3_sitelib}/%{pypi_name}-*.egg-info/
 
 %changelog
+* Tue Jun 21 2022 Sumedh Sharma <sumsharma@microsoft.com> - 8.0.4-3
+- Import from Fedora 36 (License: BSD)
```
Comment (oliviacrain) [inline SPECS/python-click/python-click.spec:61]: This should be MIT, and I believe this needs different wording to pass our checks- see the logs for details.

File: SPECS/python-click/python-click.spec [hunk 2]
```diff
@@ -30,132 +20,214 @@ comes with good defaults out of the box.
 
 %package -n     python%{python3_pkgversion}-%{pypi_name}
 Summary:        %{summary}
-%{?python_provide:%python_provide python%{python3_pkgversion}-%{pypi_name}}
+
 BuildRequires:  python%{python3_pkgversion}-devel
-BuildRequires:  python%{python3_pkgversion}-setuptools
+BuildRequires:  pyproject-rpm-macros
+BuildRequires:  python-wheel
```
Comment (oliviacrain) [inline SPECS/python-click/python-click.spec:26]:
```suggestion
BuildRequires:  python3-wheel
```

**Review by oliviacrain — CHANGES_REQUESTED** [review]

File: SPECS/python-click/python-click.spec [hunk 3]
```diff
@@ -30,132 +20,214 @@ comes with good defaults out of the box.
...
 %prep
 %autosetup -n %{pypi_name}-%{version} -p1
+# Use test dependencies without version locks
+sed -i 's|requirements/tests.txt|requirements/tests.in|' tox.ini
+# Unpin pytest version
+# https://github.com/pallets/click/commit/91b2cd67dd04b2843fd6c71e582876c826f5b78e
+sed -i 's|pytest<7|pytest|' requirements/tests.in
+
+%generate_buildrequires
+%pyproject_buildrequires -t
```
[the remark this answers is not in the document]
Comment (Sumynwa) [reply SPECS/python-click/python-click.spec]: Done

File: SPECS/python-click/python-click.spec (hunk 1 above)
Comment (Sumynwa) [reply SPECS/python-click/python-click.spec]: This line should not be there. Removed this.

File: SPECS/python-click/python-click.spec (hunk 2 above)
Comment (Sumynwa) [reply SPECS/python-click/python-click.spec]: Done.

File: SPECS/python-click/python-click.spec [hunk 4]
```diff
@@ -1,24 +1,14 @@
...
 Name:           python-%{pypi_name}
-Version:        7.1.2
-Release:        5%{?dist}
+Version:        8.0.4
+Release:        3%{?dist}
+Summary:        Simple wrapper around optparse for powerful command line utilities
 License:        BSD
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 URL:            https://github.com/mitsuhiko/click
```
[the remark this answers is not in the document]
Comment (Sumynwa) [reply SPECS/python-click/python-click.spec]: Updated.

File: SPECS/python-click/python-click.spec [hunk 5]
```diff
@@ -1,24 +1,14 @@
-%{!?python3_pkgversion: %global python3_pkgversion 3}
-%{!?python3_version: %define python3_version %(python3 -c "import sys; sys.stdout.write(sys.version[:3])")}
-%{!?python3_sitelib: %define python3_sitelib %(python3 -c "from distutils.sysconfig import get_python_lib;print(get_python_lib())")}
-%{!?__python3: %global __python3 /usr/bin/python3}
-%{!?py3_build: %define py3_build CFLAGS="%{optflags}" %{__python3} setup.py build}
-%{!?py3_install: %define py3_install %{__python3} setup.py install --skip-build --root %{buildroot}}
-
 %global pypi_name click
```
[the remark this answers is not in the document]
Comment (Sumynwa) [reply SPECS/python-click/python-click.spec:1]: Done.

File: SPECS/python-click/python-click.spec [hunk 6]
```diff
@@ -28,134 +20,219 @@ comes with good defaults out of the box.
 
 %description %{_description}
 
-%package -n     python%{python3_pkgversion}-%{pypi_name}
+%package -n     python3-%{pypi_name}
```
Comment (oliviacrain) [inline SPECS/python-click/python-click.spec:23]: This package should have a runtime requirement on `python3`

Comment (Sumynwa) [reply SPECS/python-click/python-click.spec:23]: Added.

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**